streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -197,92 +195,254 @@ public boolean isActive() {
         return true;
     }
 
+    /**
```

Comment:
@vvcephei the diff looks a bit over since there are some function reordering, but the real changes are only in suspend / resume / commit / close branching on the state.